src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -23,7 +23,13 @@ public sealed partial class JsonDocument : IDisposable
     {
         private ReadOnlyMemory<byte> _utf8Json;
         private MetadataDb _parsedData;
-        private byte[]? _extraRentedBytes;
+
+        private byte[]? _extraRentedArrayPoolBytes;
```

Comment:
Why are the extra flags needed? They seem to be equivalent to checking whether the rented references are null.

Reply:
The existing pattern was thread-safe using `Interlocked.Exchange()`. Instead of doing two calls to `Interlocked.Exchange()` (the original one and the new one), a `bool` was added that will avoid the `Interlocked.Exchange()` if not necessary.